From e50ef81380f03941bcca6340a8602a20f7295b63 Mon Sep 17 00:00:00 2001
From: Yutaka Ando <r46913@freescale.com>
Date: Sun, 19 Sep 2010 13:38:31 +0800
Subject: [PATCH] PM: P1022DS: Add the mpic timer support to dts

dts file changes to add MPIC timer as a wake up source.

Extracted from the P1022DS_20100330-ltib.iso vendor drop.

Signed-off-by: Yutaka Ando <r46913@freescale.com>
Signed-off-by: Dave Liu <daveliu@freescale.com>
Signed-off-by: Yang Shi <yang.shi@windirver.com>
---
 arch/powerpc/boot/dts/p1022ds.dts |   12 ++++++++++--
 1 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/arch/powerpc/boot/dts/p1022ds.dts b/arch/powerpc/boot/dts/p1022ds.dts
index b0846da..ea7d576 100644
--- a/arch/powerpc/boot/dts/p1022ds.dts
+++ b/arch/powerpc/boot/dts/p1022ds.dts
@@ -470,11 +470,19 @@
 
 		mpic: pic@40000 {
 			interrupt-controller;
-			#address-cells = <0>;
+			#address-cells = <1>;
+			#size-cells = <1>;
 			#interrupt-cells = <2>;
 			reg = <0x40000 0x40000>;
-			compatible = "chrp,open-pic";
+			compatible = "chrp,open-pic", "simple-bus";
 			device_type = "open-pic";
+
+			timer@1100 {
+				compatible = "fsl,mpic-global-timer";
+				reg = <0x41100 0x204>;
+				interrupts = <0xf7 0x2>;
+				interrupt-parent = <&mpic>;
+			};
 		};
 
 		msi@41600 {
-- 
1.7.0.4

